<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>807</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="pushButton_add_IPv4_blacklist">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>130</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Add</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_add_IPv4_blacklist">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>130</y>
      <width>251</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_enable_IPv4_blacklist">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>90</y>
      <width>201</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Blacklist</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup_IPv4</string>
    </attribute>
   </widget>
   <widget class="QRadioButton" name="radioButton_enable_IPv4_whitelist">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>250</y>
      <width>201</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Whitelist</string>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup_IPv4</string>
    </attribute>
   </widget>
   <widget class="QPushButton" name="pushButton_add_IPv4_whitelist">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>290</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Add</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_add_IPv4_whitelist">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>290</y>
      <width>251</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_IPv4_adress_filter">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>60</y>
      <width>201</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>IPv4 Adress Filter</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_enable_canid_blacklist">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>450</y>
      <width>201</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Blacklist</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup_canID</string>
    </attribute>
   </widget>
   <widget class="QPushButton" name="pushButton_add_canid_blacklist">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>490</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Add</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_add_canid_whitelist">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>650</y>
      <width>251</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_add_canid_blacklist">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>490</y>
      <width>251</width>
      <height>20</height>
     </rect>
    </property>
   </widget>
   <widget class="QRadioButton" name="radioButton_enable_canid_whitelist">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>610</y>
      <width>201</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Whitelist</string>
    </property>
    <property name="checked">
     <bool>false</bool>
    </property>
    <attribute name="buttonGroup">
     <string notr="true">buttonGroup_canID</string>
    </attribute>
   </widget>
   <widget class="QLabel" name="label_canid_filter">
    <property name="geometry">
     <rect>
      <x>910</x>
      <y>420</y>
      <width>201</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>CAN-ID Filter</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_add_canid_whitelist">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>650</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Add</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_rec_msg">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>0</y>
      <width>201</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Recieved Messages</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_log_msg">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>260</y>
      <width>221</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Logged Messages</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_start_stop">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>10</y>
      <width>271</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Start</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_start_recording">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>550</y>
      <width>211</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Start Recording</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_max_log_msg">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>560</y>
      <width>113</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_update_max_log_msg">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>560</y>
      <width>121</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Apply</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textBrowser_rec_msg">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>40</y>
      <width>750</width>
      <height>200</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textBrowser_log_msg">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>300</y>
      <width>750</width>
      <height>250</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textBrowser_IPv4_blacklist">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>160</y>
      <width>351</width>
      <height>81</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textBrowser_IPv4_whitelist">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>320</y>
      <width>351</width>
      <height>81</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textBrowser_canID_blacklist">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>520</y>
      <width>351</width>
      <height>81</height>
     </rect>
    </property>
   </widget>
   <widget class="QTextBrowser" name="textBrowser_canID_whitelist">
    <property name="geometry">
     <rect>
      <x>830</x>
      <y>680</y>
      <width>351</width>
      <height>81</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_sending_can_msg">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>620</y>
      <width>221</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>16</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Sending CAN-Message</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="plainTextEdit_target_ESP_IPv4_adress">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>660</y>
      <width>241</width>
      <height>25</height>
     </rect>
    </property>
    <property name="toolTip">
     <string extracomment="Bsp.&quot;192.168.0.240&quot;"/>
    </property>
    <property name="plainText">
     <string>192.168.0.240</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="plainTextEdit_target_can_id">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>690</y>
      <width>241</width>
      <height>25</height>
     </rect>
    </property>
    <property name="plainText">
     <string>69</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="plainTextEdit_hexadecimal_pair_7">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>730</y>
      <width>61</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="plainText">
     <string>00</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_ESP_IPv4_adress">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>660</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>ESP-IPv4 adress</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_canid">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>690</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>CAN ID</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_cand_data">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>740</y>
      <width>91</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>CAN data</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="plainTextEdit_hexadecimal_pair_6">
    <property name="geometry">
     <rect>
      <x>90</x>
      <y>730</y>
      <width>61</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="plainText">
     <string>00</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="plainTextEdit_hexadecimal_pair_5">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>730</y>
      <width>61</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="plainText">
     <string>00</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="plainTextEdit_hexadecimal_pair_4">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>730</y>
      <width>61</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="plainText">
     <string>00</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="plainTextEdit_hexadecimal_pair_3">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>730</y>
      <width>61</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="plainText">
     <string>00</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="plainTextEdit_hexadecimal_pair_2">
    <property name="geometry">
     <rect>
      <x>370</x>
      <y>730</y>
      <width>61</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="plainText">
     <string>00</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="plainTextEdit_hexadecimal_pair_1">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>730</y>
      <width>61</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="plainText">
     <string>00</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="plainTextEdit_hexadecimal_pair_0">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>730</y>
      <width>61</width>
      <height>30</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="plainText">
     <string>00</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_send_can_msg">
    <property name="geometry">
     <rect>
      <x>390</x>
      <y>650</y>
      <width>241</width>
      <height>71</height>
     </rect>
    </property>
    <property name="text">
     <string>Send Message</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_clear_canid_filter">
    <property name="geometry">
     <rect>
      <x>1060</x>
      <y>430</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>CLEAR</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_clear_IPv4_filter">
    <property name="geometry">
     <rect>
      <x>1100</x>
      <y>70</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>CLEAR</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_Advanced_Settings">
    <property name="geometry">
     <rect>
      <x>1110</x>
      <y>10</y>
      <width>61</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Advanced 
Settings</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
 <buttongroups>
  <buttongroup name="buttonGroup_IPv4"/>
  <buttongroup name="buttonGroup_canID"/>
 </buttongroups>
</ui>
